services/backend_service.py

```
@@ -0,0 +1,58 @@
+                json=payload.model_dump(),
+                headers=headers,
+                timeout=10.0,
+            )
+
+            response.raise_for_status()
+
+        except httpx.HTTPStatusError as e:
+            raise ValueError(
+                f"backend returned error: "
+                f"status={e.response.status_code}, body={e.response.text}"
+            )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)